src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarFile.cs

```diff
@@ -0,0 +1,274 @@
+            sourceDirectoryName = Path.GetFullPath(sourceDirectoryName);
+
+            CreateFromDirectoryInternal(sourceDirectoryName, destination, includeBaseDirectory, leaveOpen: true);
+        }
+
+        // /// <summary>
+        // /// Asynchronously creates a tar stream that contains all the filesystem entries from the specified directory.
+        // /// </summary>
+        // /// <param name="sourceDirectoryName">The path of the directory to archive.</param>
+        // /// <param name="destination">The destination stream of the archive.</param>
+        // /// <param name="includeBaseDirectory"><see langword="true"/> to include the base directory name as the first path segment in all the names of the archive entries. <see langword="false"/> to exclude the base directory name from the entry name paths.</param>
+        // /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None" />.</param>
+        // /// <returns>A task that represents the asynchronous creation operation.</returns>
+        // public static Task CreateFromDirectoryAsync(string sourceDirectoryName, Stream destination, bool includeBaseDirectory, CancellationToken cancellationToken = default)
+        // {
+        //     throw new NotImplementedException();
+        // }
+
+        /// <summary>
+        /// Creates a tar file that contains all the filesystem entries from the specified directory.
+        /// </summary>
+        /// <param name="sourceDirectoryName">The path of the directory to archive.</param>
+        /// <param name="destinationFileName">The path of the destination archive file.</param>
+        /// <param name="includeBaseDirectory"><see langword="true"/> to include the base directory name as the first path segment in all the names of the archive entries. <see langword="false"/> to exclude the base directory name from the entry name paths.</param>
+        public static void CreateFromDirectory(string sourceDirectoryName, string destinationFileName, bool includeBaseDirectory)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(sourceDirectoryName);
+            ArgumentException.ThrowIfNullOrEmpty(destinationFileName);
+
+            // Rely on Path.GetFullPath for validation of paths
+            sourceDirectoryName = Path.GetFullPath(sourceDirectoryName);
+            destinationFileName = Path.GetFullPath(destinationFileName);
+
+            if (!Directory.Exists(sourceDirectoryName))
+            {
+                throw new DirectoryNotFoundException(string.Format(SR.IO_PathNotFound_Path, sourceDirectoryName));
+            }
+
+            if (Path.Exists(destinationFileName))
+            {
+                throw new IOException(string.Format(SR.IO_FileExists_Name, destinationFileName));
+            }
+
+            using FileStream fs = File.Create(destinationFileName, bufferSize: 0x1000, FileOptions.None);
```

Comment:
if you create `FileStream` with `FileMode.CreateNew` it's going to throw if the file already exists.

```suggestion
            using FileStream fs = new (destinationFileName, FileMode.CreateNew, FileAccess.Write);
```